src/transaction.cpp

```diff
@@ -1074,7 +1074,9 @@ bool wsrep::transaction::bf_abort(
             }
 
             lock.unlock();
-            server_service_.background_rollback(client_state_);
+            /* if background rollback is skipped, reset rollbacker activity */
+            if (server_service_.background_rollback(client_state_))
+	        client_state_.set_rollbacker_active(false);
```

Comment:
Calling `set_rollbacker_active()` should be done with lock held to avoid races.

Reply:
Also, it is possible that the client session has reached `client_state::do_wait_rollback_complete_and_acquire_ownership()` while the mutex was unlocked above. In order to wake up the client, the code should also notify the `client_state` via `client_state_.cond_` to wake up the thread which hosts the client session.